### ISSUE TITLE
Production grade Dockerfile

### ISSUE THREAD

**Issue by TheAssassin** [openSUSE/osem#1377]
I am working on a Dockerfile for deploying OSEM in production at the moment. As @hennevogel mentioned in #1213, this might be considerable to add to this project.

I think that the OSEM might benefit quite much from having this included upstream, because that simplifies deployment greatly (especially since there's no installation documentation, the referral to

> tons of guides on how to deploy rails apps on various base operating systems

are not really useful to the end user, especially unexperienced people. I have struggled a bit, too, during the creation of my Dockerfile.

I would be willing to send in a PR for this, although I would not pin it down to MySQL/MariaDB, as PostgreSQL works just as fine as SQLite3 (latter is useful for people who just want to try it out, it can work smaller scale deployments, too).

**Comment by lagartoflojo** [comment]
I like the idea of creating a production-ready Dockerfile.
It would be especially cool if we can automatically create the image and push it to Docker Hub. Not sure how that all works.

> I would not pin it down to MySQL/MariaDB, as PostgreSQL works just as fine as SQLite3

I think SQLite3 is completely discarded for production usage. Yes, it's possible.. but really?
We should provide sane defaults. 
Either way, in Dockerlandia, the db should run in a separate container. So in order to spin up OSEM as easily as possible, we can provide a `docker-compose.yml` file configured to use MySQL (Postgres is [not 100% compatible yet](https://github.com/openSUSE/osem/pull/1229)), and then anyone can change it to their heart's desire, if they need to.

**Comment by TheAssassin** [comment]
> Either way, in Dockerlandia, the db should run in a separate container. So in order to spin up OSEM as easily as possible, we can provide a docker-compose.yml file configured to use MySQL (Postgres is not 100% compatible yet), and then anyone can change it to their heart's desire, if they need to.

That's the plan, yeah, I already wrote such a file with MariaDB integration and called it `docker-compose.yml.example`.

> I think SQLite3 is completely discarded for production usage. Yes, it's possible.. but really?
We should provide sane defaults.

The idea behind it is that for creating a Dockerfile or just testing, an SQLite3 database is way easier to set up at the moment. The only thing I really had to change to get SQLite3 working was moving the `sqlite3` dependency up a little bit in the Gemfile.

**Comment by lagartoflojo** [comment]
> creating a Dockerfile or just testing,

By "for just testing", do you mean to "spin up OSEM and see if it's good enough for my use case"?

**Comment by TheAssassin** [comment]
Yep, that's what I initially wanted to do before finding out there's literally no installation docs. Although I have to admit, with a `docker-compose.yml.example` file showcasing MariaDB integration, such a test would be as easy as with SQLite3.

**Comment by lagartoflojo** [comment]
Yeah, the docker-compose file would be really helpful, and if it's production-ready, even better.
Still I think that to try out OSEM, it's much easier to just go to demo.osem.io :)

**Comment by TheAssassin** [comment]
I've set it all up and created a PR (see above). I followed your words about database support and set it up to use MariaDB by default. The version in the PR solves the

> "spin up OSEM and see if it's good enough for my use case"?

problem by even including MailHog, a tiny SMTP server that intercepts every mail and displays it in a browser.

I haven't got to writing documentation about it, but I put lots of comments into the source files.

The comments also describe how to switch from MailHog to a **real** mail server, which is likely the only step that is required to make it a production setup.
